Content.Server/Labels/Label/LabelSystem.cs

```diff
@@ -50,21 +51,14 @@ public void Label(EntityUid uid, string? text, MetaDataComponent? metadata = nul
 
             if (string.IsNullOrEmpty(text))
             {
-                if (label.OriginalName is null)
-                    return;
-
                 // Remove label
-                _metaData.SetEntityName(uid, label.OriginalName, metadata);
```

Comment:
this removes the functionality of the labels changing the name of the entity, which i feel should be preserved